### ISSUE TITLE
Incorrect indenting of ternary with multi-line function calls with offsetTernaryExpressions: true

### ISSUE THREAD

**Issue by brodybits** [eslint/eslint#14058]
__updated:__

## Part 1a: Regression

**Tell us about your environment**

* **ESLint Version:** `v7.19.0`
* **Node Version:** `v14.15.4`
* **npm Version:** `6.14.10`

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

Here is my `.eslintrc.yml` configuration from <https://github.com/brodybits/react-native-module-init>, but with Prettier plugin items removed I think they are not important for this issue:

<details>
<summary>Configuration</summary>

```yaml
parserOptions:
  ecmaVersion: 9

extends: standard

plugins:
  - jest

env: { jest/globals: true }
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I am raising this with a redaction of the affected source code from <https://github.com/brodybits/react-native-module-init>, which is not formatted right by eslint `v7.19.0` as I said in <https://github.com/brodybits/react-native-module-init/pull/90#issuecomment-770473490>.

Here is the redaction of the affected source code that I have made in `await-prompt1.js`:

```js
const prompts = require('prompts')

function onPromptState ({ aborted }) {
  if (aborted) {
    process.exit(1)
  }
}

function prompt (props) {
  return prompts([{ onState: onPromptState, ...props }])
}

Promise.resolve().then(async () => {
  const platforms = ['android', 'ios']

  const { androidPackageId } =
    platforms.indexOf('android') !== -1
      ? await prompt({
          type: 'text',
          name: 'androidPackageId',
          message: 'What is the desired Android package id?',
          initial: 'com.demo',
          validate: androidPackageId => androidPackageId.length > 0
        })
      : { androidPackageId: null }

  console.log({ androidPackageId })
})
```

```bash
npx eslint await-prompt1.js
```

**What did you expect to happen?**

Accepts the formatting as-is, like eslint `7.18.0`

**What actually happened? Please include the actual, raw output from ESLint.**

```console
/Users/brodybits/dev/react-native-module-init/await-prompt1.js
  19:1  error  Expected indentation of 8 spaces but found 10  indent
  20:1  error  Expected indentation of 8 spaces but found 10  indent
  21:1  error  Expected indentation of 8 spaces but found 10  indent
  22:1  error  Expected indentation of 8 spaces but found 10  indent
  23:1  error  Expected indentation of 8 spaces but found 10  indent
  24:1  error  Expected indentation of 6 spaces but found 8   indent

✖ 6 problems (6 errors, 0 warnings)
  6 errors and 0 warnings potentially fixable with the `--fix` option.
```

When I try `npx eslint await-prompt1.js --fix`, it changes the code like this:

```diff
--- await-prompt1.js.orig	2021-01-31 19:21:20.000000000 -0500
+++ await-prompt1.js	2021-01-31 19:32:02.000000000 -0500
@@ -16,12 +16,12 @@
   const { androidPackageId } =
     platforms.indexOf('android') !== -1
       ? await prompt({
-          type: 'text',
-          name: 'androidPackageId',
-          message: 'What is the desired Android package id?',
-          initial: 'com.demo',
-          validate: androidPackageId => androidPackageId.length > 0
-        })
+        type: 'text',
+        name: 'androidPackageId',
+        message: 'What is the desired Android package id?',
+        initial: 'com.demo',
+        validate: androidPackageId => androidPackageId.length > 0
+      })
       : { androidPackageId: null }
 
   console.log({ androidPackageId })
```

**Are you willing to submit a pull request to fix this bug?**

Yes (best effort)

This is regression is blocking <https://github.com/standard/standard/issues/1624>, which could have otherwise been resolved by the recent eslint `7.19.0` update.

I think this regression is caused by my contribution in <https://github.com/eslint/eslint/pull/13972>, which I should have tested on <https://github.com/brodybits/react-native-module-init> before submitting it for review.

## Part 1b: General case - object reducer function

I have reproduced a more general manifestation of the issue with an object reducer function in `tests/lib/rules/indent.js`:

```js
        {
            code: unIndent`
              condition1
                ? reduce({
                  first,
                  second,
                })
                : reduce({
                  third,
                  fourth,
                })
            `,
            options: [2, { offsetTernaryExpressions: true }]
        },
```

If I would revert the `lib` update in e1da90f (PR #13972), then this would be the expected indentation __now unbalanced__:

```js
        {
            code: unIndent`
              condition1
                ? reduce({
                    first,
                    second,
                  })
                : reduce({
                  third,
                  fourth,
                })
            `,
            options: [2, { offsetTernaryExpressions: true }]
        },
```

## Part 2: General case - pipe function

I have reproduced another more general manifestation with a pipe function in `tests/lib/rules/indent.js`:

```js
        {
            code: unIndent`
              condition1
                ? pipe(
                  first,
                  second,
                )
                : pipe(
                  third,
                  fourth,
                )
            `,
            options: [2, { offsetTernaryExpressions: true }]
        },
```

This seems to be inconsistent with the expected formatting documented in: <https://eslint.org/docs/rules/indent#offsetternaryexpressions>

If I would revert the change to `lib` in e1da90fc414a3c9c16f52db4a5bd81bd4f9532a4 (PR #13972), then the expected indenting is shown as follows __now unbalanced__:

```js
        {
            code: unIndent`
              condition1
                ? pipe(
                    first,
                    second,
                  )
                : pipe(
                  third,
                  fourth,
                )
            `,
            options: [2, { offsetTernaryExpressions: true }]
        },
```

which looks *similar* to closed issue #13425.

**Comment by btmills** [comment]
Agreed this looks like a regression. I still have the release issue open, so while this isn’t a show-stopper that absolutely requires a patch release, if there’s a fix available tomorrow, we could release it as 7.19.1. The other option is reverting the PR that caused it and re-attempting the fix in the release two weeks from now.

**Comment by brodybits** [comment]
I would say that this issue is more than a regression.

I have now added multiple general cases to the description. I would say that reverting e1da90f (PR #13972) would lead to multiple cases of unbalanced ternary formatting:

```js
        // issue #13971:
        {
            code: unIndent`
              condition1
                ? condition2
                    ? Promise.resolve(1)
                    : Promise.resolve(2)
                : condition2
                  ? Promise.resolve(2)
                  : Promise.resolve(3)
            `,
            options: [2, { offsetTernaryExpressions: true }]
        },
        // from Part 1a in the description:
        {
            code: unIndent`
              condition1
                ? reduce({
                    first,
                    second,
                  })
                : reduce({
                  third,
                  fourth,
                })
            `,
            options: [2, { offsetTernaryExpressions: true }]
        },
        // from Part 2 in the description:
        {
            code: unIndent`
              condition1
                ? pipe(
                    first,
                    second,
                  )
                : pipe(
                  third,
                  fourth,
                )
            `,
            options: [2, { offsetTernaryExpressions: true }]
        },
```

I would personally favor taking some more time to understand what is going on and how we can resolve the root causes before simply reverting e1da90f (PR #13972).

**Comment by Tol1** [comment]
So... No updates? After upgrading I got 300+ errors about indent, so I downgraded after saw this issue. And seems like I still cannot upgrade if I don't want to run fixes back and forth

**Comment by pdeveltere** [comment]
This is also an issue for us. `offsetTernaryExpressions` is not working as expected!

**Comment by Tol1** [comment]
Woo, this issue has birthday soon. With major version release inbetween, still nothing?

**Comment by liuxy0551** [comment]
This problem still exists.

**Comment by emurano** [comment]
Still exists in v8.28.0

**Comment by jsonMartin** [comment]
This still is not working for me either.

**Comment by sam3k** [comment]
Quick summary update on this issue:

- A revert was never made to undo the changes
- The team is [awaiting](https://github.com/eslint/eslint/issues/14015#issuecomment-772241479) for a full fix to be submitted

Anyone wants to submit a PR to fix this?

**Comment by nzakas** [comment]
@matt-jolt if you'd like to submit a PR, please do. The team has a lot of higher priorities to deal with, and as such, issues like this tend to sit for a while. There's just not enough time to tackle every issue.

**Comment by nzakas** [comment]
Hi folks, I just wanted to let everyone know that the ESLint team has decided to deprecate all formatting rules, including indent, and recommend people use source code formatters like [Prettier](https://prettier.io) or [dprint](https://dprint.dev) instead. As a result, we are closing all outstanding issues related to these rules. For more information, please see https://github.com/eslint/eslint/issues/17522. Thanks.
